crates/cli/cli/src/commands/keys/list.rs

```diff
@@ -0,0 +1,68 @@
+use core::fmt::Write;
+use std::collections::BTreeMap;
+
+use oneline_eyre::eyre::eyre;
+
+use hermes_cli_framework::command::CommandRunner;
+use hermes_cli_framework::output::{json, Output};
+use hermes_cosmos_relayer::contexts::builder::CosmosBuilder;
+
+use ibc_relayer::config::{ChainConfig, Config};
+use ibc_relayer_types::core::ics24_host::identifier::ChainId;
+
+use crate::Result;
+
+#[derive(Debug, clap::Parser)]
+pub struct KeysListCmd {
+    #[clap(
+        long = "chain",
+        required = true,
+        value_name = "CHAIN_ID",
+        help_heading = "REQUIRED",
+        help = "Identifier of the chain"
+    )]
+    chain_id: ChainId,
+}
+
+#[derive(Clone, Debug)]
+pub struct KeysListOptions {
+    pub chain_config: ChainConfig,
+}
+
+impl KeysListCmd {
+    fn options(&self, config: &Config) -> Result<KeysListOptions> {
+        let chain_config = config
+            .find_chain(&self.chain_id)
+            .ok_or_else(|| eyre!("chain `{}` not found in configuration file", self.chain_id))?;
+
+        Ok(KeysListOptions {
+            chain_config: chain_config.clone(),
+        })
+    }
+}
+
+impl CommandRunner<CosmosBuilder> for KeysListCmd {
+    async fn run(&self, builder: &CosmosBuilder) -> Result<Output> {
+        let config = builder.config;
+
+        let opts = match self.options(&config) {
+            Err(e) => Output::error(e).exit(),
+            Ok(opts) => opts,
+        };
+
+        match opts.chain_config.list_keys() {
```

Comment:
I'm getting an error here that the `list_keys` function cannot be found. Looking at the `ChainConfig` type, I get the following struct:
```rust
#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
#[serde(deny_unknown_fields)]
pub struct ChainConfig {
    /// The chain's network identifier
    pub id: ChainId,

    /// The chain type
    #[serde(default = "default::chain_type")]
    pub r#type: ChainType,

    /// The RPC URL to connect to
    pub rpc_addr: Url,

    /// The gRPC URL to connect to
    pub grpc_addr: Url,

    /// The type of event source and associated settings
    pub event_source: EventSourceMode,

    /// Timeout used when issuing RPC queries
    #[serde(default = "default::rpc_timeout", with = "humantime_serde")]
    pub rpc_timeout: Duration,

    /// Whether or not the full node Hermes connects to is trusted
    #[serde(default = "default::trusted_node")]
    pub trusted_node: bool,

    pub account_prefix: String,
    pub key_name: String,
    #[serde(default)]
    pub key_store_type: Store,
    pub key_store_folder: Option<PathBuf>,
    pub store_prefix: String,
    pub default_gas: Option<u64>,
    pub max_gas: Option<u64>,

    // This field is only meant to be set via the `update client` command,
    // for when we need to ugprade a client across a genesis restart and
    // therefore need and archive node to fetch blocks from.
    pub genesis_restart: Option<GenesisRestart>,

    // This field is deprecated, use `gas_multiplier` instead
    pub gas_adjustment: Option<f64>,
    pub gas_multiplier: Option<GasMultiplier>,

    pub fee_granter: Option<String>,
    #[serde(default)]
    pub max_msg_num: MaxMsgNum,
    #[serde(default)]
    pub max_tx_size: MaxTxSize,
    #[serde(default = "default::max_grpc_decoding_size")]
    pub max_grpc_decoding_size: Byte,

    /// A correction parameter that helps deal with clocks that are only approximately synchronized
    /// between the source and destination chains for a client.
    /// This parameter is used when deciding to accept or reject a new header
    /// (originating from the source chain) for any client with the destination chain
    /// that uses this configuration, unless it is overridden by the client-specific
    /// clock drift option.
    #[serde(default = "default::clock_drift", with = "humantime_serde")]
    pub clock_drift: Duration,

    #[serde(default = "default::max_block_time", with = "humantime_serde")]
    pub max_block_time: Duration,

    /// The trusting period specifies how long a validator set is trusted for
    /// (must be shorter than the chain's unbonding period).
    #[serde(default, with = "humantime_serde")]
    pub trusting_period: Option<Duration>,

    /// CCV consumer chain
    #[serde(default = "default::ccv_consumer_chain")]
    pub ccv_consumer_chain: bool,

    #[serde(default)]
    pub memo_prefix: Memo,

    // This is an undocumented and hidden config to make the relayer wait for
    // DeliverTX before sending the next transaction when sending messages in
    // multiple batches. We will instruct relayer operators to turn this on
    // in case relaying failed in a chain with priority mempool enabled.
    // Warning: turning this on may cause degradation in performance.
    #[serde(default)]
    pub sequential_batch_tx: bool,

    // Note: These last few need to be last otherwise we run into `ValueAfterTable` error when serializing to TOML.
    //       That's because these are all tables and have to come last when serializing.
    #[serde(
        default,
        skip_serializing_if = "Option::is_none",
        with = "self::proof_specs"
    )]
    pub proof_specs: Option<ProofSpecs>,

    // These last few need to be last otherwise we run into `ValueAfterTable` error when serializing to TOML
    /// The trust threshold defines what fraction of the total voting power of a known
    /// and trusted validator set is sufficient for a commit to be accepted going forward.
    #[serde(default)]
    pub trust_threshold: TrustThreshold,

    pub gas_price: GasPrice,

    #[serde(default)]
    pub packet_filter: PacketFilter,

    #[serde(default)]
    pub address_type: AddressType,
    #[serde(default = "Vec::new", skip_serializing_if = "Vec::is_empty")]
    pub extension_options: Vec<ExtensionOption>,
    pub compat_mode: Option<CompatMode>,
    pub clear_interval: Option<u64>,
}
```
This type is located in `ibc-relayer-0.26.4/src/config.rs`. It indeed does _**not**_ have a `list_keys` function defined on it. 

Looking in the hermes codebase, the type that I need in order to access the `list_keys` method is [this](https://github.com/informalsystems/hermes/blob/master/crates/relayer/src/config.rs#L630-L632) one, located in `crates/relayer/src/config.rs`; the same file as the above incorrect `ChainConfig` type. 

Is this simply a matter of updating the `ibc-relayer` dependency? Otherwise, I'm not sure how to access the `ChainConfig` type that I need here.

Reply:
Never mind, realized that in `ibc-relayer-0.26.4` the `list_keys` function is a standalone function in the `keyring` module.